infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -734,4 +769,46 @@ public void shutdown() {
     private String getLocalHost() {
         return NodeLocator.parseString(serverContext.getLocalEndpoint()).getHost();
     }
+
+    private LockClient createLockClient(LockStateType initState) throws Exception {
+        LockConfig lockConfig = createLockConfig(serverContext.getLogReplicationConfig());
+        LockClient lockClient =
+                new LockClient(logReplicationNodeId,
+                        lockConfig, runtime.get());
+
+        LockListener lockListener =
+                new LogReplicationLockListener(this);
+
+        LockDataTypes.LockId lockId = LockDataTypes.LockId.newBuilder()
```

Comment:
From L782 to L791, isn't this what LockClient.registerInterest does? (without the monitorLock part). Would it be cleaner for it to be part of LockClient instead? 
Also, can we add some comments as to why we need not to do the monitoring in this case?

Reply:
Ideally, LockClient would need to have a static factory where it's being created and returned to a caller. However, LockClient relies on a LockReplicationListener which is a part of infra package and also depends on the DiscoveryService. Since, Java does not allow circular dependencies this is not possible without major refactoring.

Reply:
ok